test/test_sssd_conf.py

```diff
@@ -0,0 +1,189 @@
+
+    :param file1:  first of compared files
+    :param file2:  second of compared files
+    :return: True if content of files matches, False otherwise
+    """
+    dict1 = parser_object_to_dict(load_file_to_parser(file1))
+    dict2 = parser_object_to_dict(load_file_to_parser(file2))
+    assert dict1 == dict2
+
+
+# TESTS
+def test_create_template_exists():
+    """
+    Template file exists
+    """
+    assert SSSDConf._template.is_file()
+
+
+def test_check_backups(tmpdir):
+    """
+    Exception is raised if backup file exists
+    """
+    tmpfile = Path(tmpdir, "testfile")
+    with tmpfile.open('w') as original:
+        original.write('Testing backup original')
+    sssd = SSSDConf()
+    sssd._backup_default = tmpfile
+    sssd._backup_original = tmpfile
+    with pytest.raises(FileExistsError, match=f'{tmpfile} file exists'):
+        sssd.check_backups()
+
+
+def test_create_content_backup_original(sssd_test_prepare):
+    """
+    Backup of original config file is created if the file exists
+    Backup of library default conf_file is created
+    """
+    sssd_test = sssd_test_prepare
+
+    # create conf
+    shutil.copy2(SSSDConf._template, sssd_test._conf_file)
+    sssd_test.create()
+
+    assert sssd_test._backup_original.exists()
+    compare_conf_files(SSSDConf._template, sssd_test._backup_original)
+    assert sssd_test._backup_default.exists()
+    compare_conf_files(SSSDConf._template, sssd_test._backup_default)
+
+
+def test_create_content_backup_default(sssd_test_prepare):
+    """
+    Backup of library default config file is always created
+    regardless of previous existence of config file
+    """
+    sssd_test = sssd_test_prepare
+
+    sssd_test.create()
+    assert sssd_test._backup_default.exists()
+    compare_conf_files(SSSDConf._template, sssd_test._backup_default)
+
+
+def test_create_content(sssd_test_prepare):
+    """
+    Content of config file is created based on template if the file is not
+    present in the system. Content of created file corresponds to template values.
+    """
+    sssd_test = sssd_test_prepare
+
+    sssd_test.create()
+
+    # content of created config is saved to backup_default and therefore
+    # it can be verified as follows
+    compare_conf_files(SSSDConf._template, sssd_test._backup_default)
+
+
+def test_create_conf_updated(sssd_test_prepare):
+    """
+    Content of config file is updated with values from template
+    if the file already exists
+    """
+    # create content of config based on test config file; load content to dict
+    sssd_test = sssd_test_prepare
+    sssd_test._conf_file = Path(FILES_DIR, "sssd_conf_test_start")
+
+    sssd_test.create()
+
+    # load reference content from library file to parser object
+    reference_file = Path(FILES_DIR, "sssd_conf_test_result")
+    reference_content = load_file_to_parser(reference_file)
+
+    assert parser_object_to_dict(sssd_test._default_parser) == parser_object_to_dict(reference_content)
```

Comment:
For this test I would suggest using different files for "original" content and for template content so we can check that the final content is really merged, but not that the original content is wiped and the template is used

Reply:
This is actually ensured by existence of 3 files. Two of them are _sssd_conf_test_start_  - original file and _sssd_conf_test_result_ - reference file after merge. Template is the 3rd file (hardcoded in SSSDConf class) that is merged into _sssd_conf_test_start_ to get _sssd_conf_test_result_ .

Reply:
Ok, get it. Resolved.